### PR TITLE
Check if buckets are outdated in `scoop status`

### PR DESCRIPTION
**Pull request by nimzo6689**
https://github.com/lukesampson/scoop/issues/3549

### CODE REVIEW

**Comment by Ash258** [comment]
Dupliacting code from update is not key.

completely easy, non resource consuming fix could look like this:,

```powershell
Test-UpdateStatus {
    Get-LocalBucket | Foreach-Object {
          $loc = Find-BucketDirectory $_ -Root
          Push-Location $loc
          git_fetch -q origin
          if ((git_status) -like 'is behind') { return $true }
    }
}
```

**Comment by nimzo6689** [comment]
@Ash258 
Thanks for the advice.
I've updated some codes for refactoring.

*In PowerShell, `return` in Foreach-Object behaves like the continue statement. (I missed it.)
So I replaced it with foreach statement for return early.

How about this?

File: libexec/scoop-status.ps1 [hunk 1]
```diff
@@ -10,25 +10,37 @@
...
-    Push-Location $currentdir
+function Test-GitRepoLatest($location) {
+    Push-Location $location
     git_fetch -q origin
-    $commits = $(git log "HEAD..origin/$(scoop config SCOOP_BRANCH)" --oneline)
-    if($commits) { $needs_update = $true }
+    $is_behind = (git_status) -like '*is behind*' -as [bool]
     Pop-Location
+
+    return !($is_behind)
```
[the remark this answers is not in the document]
Comment (r15ch13) [reply libexec/scoop-status.ps1]: Also renaming to `Test-IsGitRepositoryBehind` would make the function more clear.

File: libexec/scoop-status.ps1 [hunk 2]
```diff
@@ -10,25 +10,37 @@
...
+        }
+    }
+
+    return $true
 }
 
-if($needs_update) {
+# check if scoop needs updating
+$needs_update = !(Test-ScoopLatest)
+if ($needs_update) {
```
[the remark this answers is not in the document]
Comment (nimzo6689) [reply libexec/scoop-status.ps1]: `$needs_update` is also used in L115.

File: libexec/scoop-status.ps1 [hunk 3]
```diff
@@ -10,25 +10,37 @@
...
+    $is_behind = (git_status) -like '*is behind*' -as [bool]
     Pop-Location
+
+    return !($is_behind)
 }
-else {
-    $needs_update = $true
+
+function Test-ScoopLatest {
+    $currentdir = fullpath $(versiondir 'scoop' 'current')
```
[the remark this answers is not in the document]
Comment (nimzo6689) [reply libexec/scoop-status.ps1]: Done.

File: libexec/scoop-status.ps1 (hunk 1 above)
[the remark this answers is not in the document]
Comment (nimzo6689) [reply libexec/scoop-status.ps1]: Done.

**Pull request closed without merging**